### ISSUE TITLE
Right oled doesnt work

### ISSUE THREAD

**Issue by redmasters** [mctechnology17/zmk-config#2]
Hi!
First, thanks for sharing your configuration with dongle support, it work flawlessly.

I've forked and added some personal configs and i have some issues with oleds:

Left peripheral:
Oled doenst show current layer active;

Right peripheral:
Oled doesnt turn on;

Here the configs:
corne_left_peripheral.conf
```c
# Oled
CONFIG_ZMK_DISPLAY=y
CONFIG_ZMK_WIDGET_LAYER_STATUS=y
CONFIG_ZMK_WIDGET_BATTERY_STATUS_SHOW_PERCENTAGE=y

# Increase the bluetooth transmission power
CONFIG_BT_CTLR_TX_PWR_PLUS_8=y
CONFIG_ZMK_BLE_EXPERIMENTAL_FEATURES=y

### MOUSE
CONFIG_ZMK_MOUSE=y

### POWER OPTIONS
# CONFIG_ZMK_EXT_POWER=y
# CONFIG_ZMK_SLEEP=n
# 1800000 milisegundos = 30 minutos * 60 segundos * 1000
# CONFIG_ZMK_IDLE_SLEEP_TIMEOUT=3600000
# Número de milisegundos de inactividad antes de entrar en estado inactivo
# CONFIG_ZMK_IDLE_TIMEOUT=60000

# Enable and adjust nice!view
# Disable other widgets, because don't work with pheripherals
# CONFIG_ZMK_WIDGET_WPM_STATUS=y
# CONFIG_ZMK_WIDGET_LAYER_STATUS=y
# CONFIG_ZMK_WIDGET_OUTPUT_STATUS=n

# # Disable in-built nice!view widget
# CONFIG_ZMK_DISPLAY_STATUS_SCREEN_BUILT_IN=y
# # Enable custom nice!view widget

### Improves compatibility with iPad
CONFIG_ZMK_HID_CONSUMER_REPORT_USAGES_BASIC=y

### HID para nice!view ePaperDisplay
# Enable eager debouncing
# "Eager Debouncing"
# Trying to lower the input lag.
CONFIG_ZMK_KSCAN_DEBOUNCE_PRESS_MS=1
CONFIG_ZMK_KSCAN_DEBOUNCE_RELEASE_MS=5
```

corne_right.conf
```c
# Oled
CONFIG_ZMK_DISPLAY=y
CONFIG_ZMK_WIDGET_LAYER_STATUS=y
CONFIG_ZMK_WIDGET_BATTERY_STATUS_SHOW_PERCENTAGE=y

# Increase the bluetooth transmission power
CONFIG_BT_CTLR_TX_PWR_PLUS_8=y
CONFIG_ZMK_BLE_EXPERIMENTAL_FEATURES=y

### MOUSE
CONFIG_ZMK_MOUSE=y

### POWER OPTIONS
# CONFIG_ZMK_EXT_POWER=y
# CONFIG_ZMK_SLEEP=n
# 1800000 milisegundos = 30 minutos * 60 segundos * 1000
# CONFIG_ZMK_IDLE_SLEEP_TIMEOUT=3600000
# Número de milisegundos de inactividad antes de entrar en estado inactivo
# CONFIG_ZMK_IDLE_TIMEOUT=60000

# Enable and adjust nice!view
# Disable other widgets, because don't work with pheripherals
# CONFIG_ZMK_WIDGET_WPM_STATUS=y
# CONFIG_ZMK_WIDGET_LAYER_STATUS=y
# CONFIG_ZMK_WIDGET_OUTPUT_STATUS=n

# # Disable in-built nice!view widget
# CONFIG_ZMK_DISPLAY_STATUS_SCREEN_BUILT_IN=y
# # Enable custom nice!view widget

### Improves compatibility with iPad
CONFIG_ZMK_HID_CONSUMER_REPORT_USAGES_BASIC=y

### HID para nice!view ePaperDisplay
# Enable eager debouncing
# "Eager Debouncing"
# Trying to lower the input lag.
CONFIG_ZMK_KSCAN_DEBOUNCE_PRESS_MS=1
CONFIG_ZMK_KSCAN_DEBOUNCE_RELEASE_MS=5
```

I've tried several ways changing the configs but no success

**Comment by redmasters** [comment]
The dongle is connected via USB-C to the computer, and even when I connect it to another device via Bluetooth, I can’t switch profiles to use the other connected device.

**Comment by mctechnology17** [comment]
You have to activate (switch) the Bluetooth every time you connect the dongle to any USB-C, be it a computer or power source and then select the saved profile or select a new profile. In the dongle configuration, this option is chosen by default.

You can change this behavior by modifying this line: https://github.com/redmasters/zmk-config-dongle/blob/633decebaa59303283610b7b3d32b8c8e2c12d12/boards/shields/corne/Kconfig.defconfig#L9C1-L9C15

In short: Without modifying anything, you just need to press `&out OUT_BLE` first to change the output and then the profile of your liking `&bt BT(0)` etc. Remember to do it every time because by default every time you plug the dongle back in the default behavior is USB!

**Comment by mctechnology17** [comment]
According to what you observe in your configuration, you do not have the output key mapped, if you want to use the dongle option, I recommend that you map it: `&out OUT_USB` `&out OUT_BLE` `&out OUT_TOG` to have versatility in the changes at convenience

**Comment by redmasters** [comment]
I understand now, I overlooked that. Thank you very much.

Any clues about the OLEDs?

**Comment by mctechnology17** [comment]
Regarding the oleds: have you tried using the reset files on the halves? You can also try enabling direct power after resetting the halves and flashing the firmware. https://github.com/redmasters/zmk-config-dongle/blob/633decebaa59303283610b7b3d32b8c8e2c12d12/config/corne.conf#L42

**Comment by mctechnology17** [comment]
You can also take a look at this config file configuration and replicate as you like, I assure you that I tested this configuration very well and it works very well https://github.com/mctechnology17/zmk-config/blob/power/domains-and-device-runtime/config/corne.conf

**Comment by redmasters** [comment]
> You can also take a look at this config file configuration and replicate as you like, I assure you that I tested this configuration very well and it works very well https://github.com/mctechnology17/zmk-config/blob/power/domains-and-device-runtime/config/corne.conf

Ill make the changes and soon return with results

**Comment by redmasters** [comment]
> Regarding the oleds: have you tried using the reset files on the halves? You can also try enabling direct power after resetting the halves and flashing the firmware. https://github.com/redmasters/zmk-config-dongle/blob/633decebaa59303283610b7b3d32b8c8e2c12d12/config/corne.conf#L42

Yes, I've used reset settings to reset the controllers every time I flash new configs

**Comment by redmasters** [comment]
Flashed out new settings, resetting all controllers with reset settings;

✅  I've mapped out the configs for output, and it works;
❌  I tried the settings for right oled and still not working;

Commit: https://github.com/redmasters/zmk-config-dongle/commit/6a7204bdc428792e8efe245321ba4f7876a7d9b6

**Comment by erasmocjunior** [comment]
The mouse movements only work on the device where the dongle is plugged in. It seems that they don’t work on devices via Bluetooth

**Comment by mctechnology17** [comment]
I just tested the dongle in Bluetooth as you describe and in OLED screen mode, it works perfectly for me. I see that your configuration is quite changed from the original repository that made the fork, I recommend that you review your configuration to have it work correctly. I also recommend using the initial configuration of this repository WITHOUT its modifications and see if it works, as I mentioned before, I have just tried all the problems that you mentioned about the repository and in no case do I have any problems.

**Comment by redmasters** [comment]
Ok, Ill try, thx for help me 🙏🏿

**Comment by mctechnology17** [comment]
I'm going to close this issue because the problem itself is its modifications and not the repository. I hope you manage to get the configuration you want, have a good day
